ratel/src/lexer/labels.rs

```diff
@@ -6,6 +6,23 @@ macro_rules! match_label {
         if $(
             $lex.next_byte() == $byte &&
         )* {$lex.bump(); !util::legal_in_label($lex.read_byte())} {
+            
+            if $token == Function {
```

Comment:
Token matching should only happen inside the parser, in this particular case here:

https://github.com/ratel-rust/ratel-core/blob/9291de29747be111a4a44a9eba947f48b6170315/ratel/src/parser/function.rs#L62-L77

At this point the `Function` token has been consumed, so matching against `MultiplicationOperator` before line 69 and then setting a flag* on the struct should be the preferential way of doing things.

\* more on that in another comment.